build/constraints.js

```diff
@@ -849,7 +928,189 @@ function splitNMCmin(sTree,pTree)
 		}
 	}
 	return vcount;
-};/********************
+};/* Equal Strength Boundaries Constraints
+ * as proposed by Nick Kalivoda
+ * assign violations if a phonological terminal is at the left/right edge of a
+ * different number of syntactic constituents as phonological constituents
+ * specific functions defined below, base function counts number of left/right
+ * edges a terminal falls on.
+ */
+
+ //this code isn't robust for terminal order-shifting, does it need to be?
+
+function equalStrengthBase(stree, ptree, scat, edgeName){
+  var sTerminals = getLeaves(stree); //syntactic terminals
+  var pTerminals = getLeaves(ptree); //prosodic terminals
+
+  /* edges refers to left or right edges, so it should be cleared out in case
+   * the edge is different from the last call */
+  for (var i = 0; i < sTerminals.length; i ++){
+    //clear out edges for syntactic terminals
+    sTerminals[i].edges = void(0);
+  }
+  // GEN also re-uses subtrees, so edges must be cleared out every time
+  for (var i = 0; i < pTerminals.length; i ++){
+    //clear out edges for prosodic terminals
+    pTerminals[i].edges = void(0);
+  }
+
+  equalStrengthHelper(stree, scat, edgeName);//call recursive helper for syntax
+  var pcat = reversableCatPairings(scat); // the prosodic cat the corresponds to scat
+  equalStrengthHelper(ptree, pcat, edgeName);//call recursive helper for prosody
+
+  //return an array of the terminals, also arrays, now with the property edges
+  return [getLeaves(stree), getLeaves(ptree)];
+}
+
+/* equalStrengthBase is run once per call so that sTerminals and pTerminals can
+ * be cleared out once. equalStrengthHelper runs recursively, once for each node
+ * on the tree. Also, the same process needs to be run on ptree and stree, so
+ * the helper function minimizes code repitition by taking only one tree and
+ * being called for both ptree and stree.
+ */
+function equalStrengthHelper(tree, cat, edgeName){
+  var boundaries = 0;
+  recursiveStrengthHelper(tree, cat, boundaries);
+  //recursiveStrengthHelper only goes through the left/right branches of tree
+  //arguments given unique names to avoid scope problems
+  function recursiveStrengthHelper(partree, category, boundaries){
+    // argument boundaries keeps track of the number of edges a terminal is on
+    var edgeIndex; // the index of the l/r edge
+    //left = [0]
+    if (edgeName == "left"){
+      edgeIndex = 0;
+    }
+    //right = [length -1]
+    if (edgeName == "right" && partree.children && partree.children.length){
+      edgeIndex = (partree.children.length - 1);
+    }
+    //increment boundaries when partree is of the correct category
+    if (partree.cat === category){
+      boundaries ++;
```

Comment:
Conventionally, there is no space between the variable name and the ++.

Reply:
oh, well I tried to change some of them but I have been putting spaces in everywhere so I didn't change all of them. Thanks for the info, though